### PR TITLE
feat(ui) Add new embedded profile to be displayed in extension

### PR DESCRIPTION
**Pull request by chriscollins3456**
This PR adds a new EmbeddedProfile that will be displayed in an iFrame of our new chrome extension. This profile reuses lots of components from an entity profile sidebar and updates those components to allow them to be readOnly (this refactoring is the cause for a good chunk of the diff). This also creates a new `UpstreamHealth` component that fetches all upstreams and their assertions so that we can look through and see if anything is failing.

**When there are failing upstream assertions:**
image

**When there is at least one passing upstream assertion and none failing:**
image

**When there are no assertions upstream or no upstreams at all:**
image

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by aditya-radhakrishnan — APPROVED** [review]
LGTM! Reviewed offline :)

**Comment by jjoyce0510** [comment]
Seems that some build is failing :)

File: datahub-web-react/src/app/entity/shared/components/styled/ExpandedOwner/OwnerContent.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { getDescriptionFromType, getNameFromType } from '../../../containers/profile/sidebar/Ownership/ownershipUtils';
+
+interface Props {
+    name: string;
+    owner: Owner;
+    hidePopOver?: boolean;
+    pictureLink?: string;
+}
+
+export default function OwnerContent({ name, owner, hidePopOver, pictureLink }: Props) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/components/styled/ExpandedOwner/OwnerContent.tsx:14]: nice refactor!

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/AboutSection/EmptyContentSection.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    const refetch = useRefetch();
+
+    return (
+        <EmptyContentWrapper>
+            <>
+                <EmptyContentMessage type="secondary">
+                    {EMPTY_MESSAGES.documentation.title}. {EMPTY_MESSAGES.documentation.description}
+                </EmptyContentMessage>
+                {!readOnly && (
+                    <Button onClick={() => routeToTab({ tabName: 'Documentation', tabParams: { editing: true } })}>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/AboutSection/EmptyContentSection.tsx:38]: Nice!

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/AboutSection/LinksSection.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+const AddLinksWrapper = styled.div`
+    margin-left: -15px;
+`;
+
+interface Props {
+    hideLinksButton?: boolean;
+    readOnly?: boolean;
+}
+
+export default function LinksSection({ hideLinksButton, readOnly }: Props) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/AboutSection/LinksSection.tsx:16]: Is this just a file being moved?

Comment (chriscollins3456) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/AboutSection/LinksSection.tsx:16]: yup! just pulled it out from the old bigger SidebarAboutSection component.

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/AboutSection/LinksSection.tsx:16]: nice!

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/LinkButton.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        white-space: nowrap;
+        text-overflow: ellipsis;
+    }
+`;
+
+interface Props {
+    link: InstitutionalMemoryMetadata;
+}
+
+export default function LinkButton({ link }: Props) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/LinkButton.tsx:25]: Awesomee!!!

File: datahub-web-react/src/app/entity/shared/containers/profile/useGetDataForProfile.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    ) => QueryResult<
+        T,
+        Exact<{
+            urn: string;
+        }>
+    >;
+    getOverrideProperties: (T) => GenericEntityProperties;
+}
+
+export default function useGetDataForProfile<T>({ urn, entityType, useEntityQuery, getOverrideProperties }: Props<T>) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/useGetDataForProfile.ts:26]: Oh thank GOD!

File: datahub-web-react/src/app/entity/shared/embed/EmbeddedProfile.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+                lineage: undefined,
+            }}
+        >
+            {loading && (
+                <LoadingWrapper>
+                    <LoadingOutlined />
+                </LoadingWrapper>
+            )}
+            {!loading && entityData && (
+                <>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/embed/EmbeddedProfile.tsx:72]: Uhm yes please

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/embed/EmbeddedProfile.tsx:72]: Look at this clean stack

File: datahub-web-react/src/app/shared/tags/TagTermGroup.tsx [hunk 7]
```diff
@@ -84,105 +64,16 @@ export default function TagTermGroup({
...
-                }
-            },
-            onCancel() {},
-            okText: 'Yes',
-            maskClosable: true,
-            closable: true,
-        });
-    };
-
-    const removeTerm = (termToRemove: GlossaryTermAssociation) => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/TagTermGroup.tsx:138]: How one thing I think almost 100% we will face merge back issues due to proposals. Just calling out!

File: datahub-web-react/src/app/shared/tags/term/StyledTerm.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    entityUrn?: string;
+    entitySubresource?: string;
+    canRemove?: boolean;
+    readOnly?: boolean;
+    highlightText?: string;
+    onOpenModal?: () => void;
+    refetch?: () => Promise<any>;
+}
+
+export default function StyledTerm(props: Props) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/term/StyledTerm.tsx:30]: Nice!

File: datahub-web-react/src/app/shared/tags/term/TermContent.tsx [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+                }
+            },
+            onCancel() {},
+            okText: 'Yes',
+            maskClosable: true,
+            closable: true,
+        });
+    };
+
+    return (
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/term/TermContent.tsx:72]: Wowwww!

File: datahub-web-react/src/graphql/search.graphql [hunk 10]
```diff
@@ -781,6 +781,17 @@ fragment searchAcrossRelationshipResults on SearchAcrossLineageResults {
     searchResults {
         entity {
             ...searchResultFields
+            ... on Dataset {
+                assertions @include(if: $includeAssertions) {
```
Comment (jjoyce0510) [inline datahub-web-react/src/graphql/search.graphql:785]: Thanks for including this conditionally

**Review by jjoyce0510 — APPROVED** [review]
LGTM. I want to be careful about the merge-back process, however, given the scope of the changes.

**Pull request merged**